### PR TITLE
chore(examples): Update all golang examples and split out golang-push tracing

### PR DESCRIPTION
**Pull request by simonswine**
* Split out tracing, from the tempo example into dedicated golang push
* Add all examples to go mod tidy and update go libraries 

Things I checked are:

Explore (classic) -> Tempo
* Can I find trace in Explore (classic) -> Tempo?
* Do I see flamegraphs within the trace spans? (Picture 2)
* Does the span to profiles link work

Explore > Profiles
* Do we have 3 services (pyroscope, loadgen and rideshare) ?
* Is the rideshare service having region, span_name and vehice labels? (Picture 1)
* Is car significantly slower than bike & scooter  (Picture 1)

### CODE REVIEW

File: examples/tracing/golang-push/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Golang Example
```
Comment (marcsanmi) [inline examples/tracing/golang-push/README.md:1]: I like this concise README.md. However, I followed the approach of writing a more detailed (likely unnecessary) one following java's approach.

I think we should make all READMEs in the `/tracing` directory consistent. I can update both my example and the Java one to follow this simpler approach, wdyt?

**Review by marcsanmi — APPROVED** [review]
LGMT 👍

File: examples/tracing/golang-push/README.md (hunk 1 above)
Comment (aleks-p) [reply examples/tracing/golang-push/README.md:1]: Fully agree!

Though perhaps this is a bit too short and we want to explain what this example is for.

**Review by kolesnikovae — APPROVED** [review]
LGTM!

Also, we need to update the README in:
* https://github.com/grafana/otel-profiling-go
* https://github.com/grafana/dskit/tree/main/spanprofiler

as they refer to the old path

File: examples/tracing/golang-push/README.md (hunk 1 above)
Comment (kolesnikovae) [reply examples/tracing/golang-push/README.md:1]: +1

Comment (simonswine) [reply examples/tracing/golang-push/README.md:1]: Yeah I was not planning to keep it like that, but I also did't necessary like the other one 😂  Fill follow up with a PR

**Pull request merged**